cpp/src/Slice/Grammar.y

```diff
@@ -1346,7 +1346,7 @@ struct_data_member
     TypeStringTokPtr ts = TypeStringTokPtr::dynamicCast($2);
     StructPtr st = StructPtr::dynamicCast(unit->currentContainer());
     assert(st);
-    $$ = st->createDataMember(ts->v.second, ts->v.first, false, 0, 0, "", ""); // Dummy
+    $$ = st->createDataMember(ts->v.second, ts->v.first, false, 0); // Dummy
```

Comment:
It's because we have useless code in the grammar that I got caught up on...
In the grammar, we have a definition for `data_member` and one specifically for `struct_data_member`, where the struct one is only for structs, and the normal one is for classes, and exceptions. But we left all the logic for struct data members in the normal `data_member` definition:
```
data_member
// ----------------------------------------------------------------------
: tagged_type_id
{
    TaggedDefTokPtr def = TaggedDefTokPtr::dynamicCast($1);
    ClassDefPtr cl = ClassDefPtr::dynamicCast(unit->currentContainer());
    DataMemberPtr dm;
    if(cl)
    {
        dm = cl->createDataMember(def->name, def->type, def->isTagged, def->tag);
    }
    StructPtr st = StructPtr::dynamicCast(unit->currentContainer());
    if(st)
    {
        dm = st->createDataMember(def->name, def->type, def->isTagged, def->tag);
    }
    ExceptionPtr ex = ExceptionPtr::dynamicCast(unit->currentContainer());
    if(ex)
    {
        dm = ex->createDataMember(def->name, def->type, def->isTagged, def->tag);
    }
    unit->currentContainer()->checkIntroduced(def->name, dm);
    $$ = dm;
```
So either I can remove all the logic from `data_member` and also tag related parameters from the `Struct::createDataMember` definition, or we can drop `struct_data_member`.

Personally, I don't see the point of having a whole separate definition for `struct_data_member`... It seems easier to me just keep everything in `data_member`, and then `Struct::createDataMember` can just issue an error if `tagged` is ever true. Or maybe there's other reason for separating the two out that I'm not seeing...

Reply:
fine, just remove any tagged logic from `Struct::createDataMember`  and assert it is not called with a tagged member